include/PathResolver.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (C) 2017-2018  Waterloo Quantitative Consulting Group, Inc. 
+// See COPYING and LICENSE files at project root for more details. 
+
+/**
+ * @file PathResolver.hpp
+ * @brief Defines the PathResolver class that allows for the resolution of paths
+ * relative to the project root path.
+ */
+
+#pragma once
+#include <string>
+#include <vector>
+#include <memory>
+#include <iostream>
+#include <whereami.h>
+#include <boost/algorithm/string.hpp>
+
+using boost::algorithm::find_all;
+using boost::algorithm::join;
+using boost::algorithm::replace_last;
+using boost::algorithm::replace_first;
+using boost::algorithm::trim_right;
+using boost::iterator_range;
+using std::string;
+using std::vector;
+using std::cout;
+using std::endl;
+using std::numeric_limits;
+using std::abs;
+using std::shared_ptr;
+using std::unique_ptr;
+
```

Comment:
You should always avoid any using statements in a header file and instead use only fully qualified names.

Reply:
D'oh, fixed.